blaze/compute/core.py

```diff
@@ -407,6 +407,29 @@ def compute(expr, d, **kwargs):
     if post_compute_:
         result = post_compute_(expr3, result, scope=d4)
 
+    if return_type == 'native':
+        # NOP, but we check for this case anyway.
+        result = result
```

Comment:
this should just be `pass`, or the very last `else` could be `elif return_type != 'native'`